### PR TITLE
Use GNU ld on m68k-unknown-linux-gnu

### PR DESCRIPTION
**Pull request by knickish**
LLD does not really support the M68k architecture yet, specify `m68k-linux-gnu-ld` as the linker for the platform

### CODE REVIEW

**Comment by workingjubilee** [comment]
@knickish Can you, in the platform support page, update m68k with the asterisk that says it supports no_std, right now?

**Comment by workingjubilee** [comment]
It currently is a question mark, which basically says "lol we're pretty sure this doesn't work", and with this it seems you've finally made it to `no_std` :3

**Comment by workingjubilee** [comment]
> @workingjubilee Building core depends on llvm patches that aren't all backported or merged yet, does that change how you'd like that documentation updated at all?

Ooh good point, I forgot about that.

Hmm... we'll revisit this next PR. :3 

@bors r+ rollup

File: compiler/rustc_target/src/spec/targets/m68k_unknown_linux_gnu.rs [hunk 1]
```diff
@@ -12,11 +12,18 @@ pub(crate) fn target() -> Target {
             description: Some("Motorola 680x0 Linux".into()),
             tier: Some(3),
             host_tools: Some(false),
-            std: None, // ?
+            std: Some(false), // std cannot be linked for now
```
Comment (glaubitz) [inline compiler/rustc_target/src/spec/targets/m68k_unknown_linux_gnu.rs:15]: I think this should be a separate commit as it's an unrelated change.

File: compiler/rustc_target/src/spec/targets/m68k_unknown_linux_gnu.rs [hunk 2]
```diff
@@ -12,11 +12,18 @@ pub(crate) fn target() -> Target {
...
         pointer_width: 32,
         data_layout: "E-m:e-p:32:16:32-i8:8:8-i16:16:16-i32:16:32-n8:16:32-a:0:16-S16".into(),
         arch: "m68k".into(),
-        options: TargetOptions { endian: Endian::Big, mcount: "_mcount".into(), ..base },
+        options: TargetOptions {
+            endian: Endian::Big,
+            mcount: "_mcount".into(),
+
+            // LLD currently does not have support for M68k
+            linker: Some("m68k-linux-gnu-ld".into()),
```
Comment (bjorn3) [inline compiler/rustc_target/src/spec/targets/m68k_unknown_linux_gnu.rs:25]: Something like `link_self_contained: LinkSelfContainedDefault::False` instead may work too. This will also ensure that attempting to enable self-contained linking using lld on the cli gives an error.

Comment (workingjubilee) [reply compiler/rustc_target/src/spec/targets/m68k_unknown_linux_gnu.rs:25]: Ah, thank you!

File: compiler/rustc_target/src/spec/targets/m68k_unknown_linux_gnu.rs (hunk 1 above)
Comment (knickish) [reply compiler/rustc_target/src/spec/targets/m68k_unknown_linux_gnu.rs]: done

File: compiler/rustc_target/src/spec/targets/m68k_unknown_linux_gnu.rs (hunk 2 above)
Comment (glaubitz) [reply compiler/rustc_target/src/spec/targets/m68k_unknown_linux_gnu.rs:25]: @knickish Can you try this?

**Comment by glaubitz** [comment]
Also, the title should be "Use GNU ld on m68k-unknown-linux-gnu" as this just applies to m68k on Linux.

**Comment by workingjubilee** [comment]
we don't have any other m68k targets tho'

File: compiler/rustc_target/src/spec/targets/m68k_unknown_linux_gnu.rs (hunk 2 above)
Comment (knickish) [reply compiler/rustc_target/src/spec/targets/m68k_unknown_linux_gnu.rs:25]: Yup, doing a build right now

Comment (knickish) [reply compiler/rustc_target/src/spec/targets/m68k_unknown_linux_gnu.rs]: This worked, `core` and everything besides `std` built and linked; the `std` link failed with the exact same error as with `linker: Some("m68k-linux-gnu-ld".into())`.

**Comment by glaubitz** [comment]
> On Nov 12, 2024, at 8:31 PM, Jubilee ***@***.***> wrote:
> 
> ﻿
> we don't have any other m68k targets tho'
> 
We could easily add m68k-unknown-elf for embedded systems and I am certainly planning that for the future as it’s useful for video game consoles such as the Sega Megadrive.

**Comment by workingjubilee** [comment]
Ah, sure then!

@bors r+ rollup
